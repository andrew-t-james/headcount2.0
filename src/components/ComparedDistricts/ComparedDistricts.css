.comparison-container {
  align-items: center;
  border: 1px solid orchid;
  display: flex;
  height: calc(100vh - 7rem);
  justify-content: space-between;
}

@media screen and (max-width: 1200px) {
  .comparison-container {
    flex-direction: column;
    justify-content: space-around;
  }
}