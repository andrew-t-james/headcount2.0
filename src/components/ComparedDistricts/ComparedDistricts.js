import React from 'react';
import PropTypes from 'prop-types';
import Card from '../Card/Card';

const ComparedDistricts = ({ comparedDistricts }) => {
  comparedDistricts = [
    {
      location: 'hellow',
      stats: {
        2000: 0.292
      }
    },
    {
      location: 'he',
      stats: {
        2000: 0.123
      }
    }
  ];
  const displaySelectedCards = comparedDistricts.map(district =>
    <article>
      <h2>{district.location}</h2>
      <p>{district.stats[2000]}</p>
    </article>
  );
  return (
    <section>
      {displaySelectedCards[0]}
      <p>District Data</p>
      {displaySelectedCards[1]}
    </section>
  );
};

ComparedDistricts.propTypes = {
  comparedDistricts: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default ComparedDistricts;