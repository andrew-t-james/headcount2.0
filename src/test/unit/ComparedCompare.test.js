import React from 'react';
import { shallow } from 'enzyme';
import ComparedDistricts from '../../components/ComparedDistricts/ComparedDistricts';

describe('ComparedDistrict unit test suite', () => {
  let wrapper;
  const mockDistricts = [
    {
      location: 'Adams',
      stats: {
        2004: 0.393
      }
    }
  ];
  beforeEach(() => wrapper = shallow(<ComparedDistricts comparedDistricts={mockDistricts}/>));

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
