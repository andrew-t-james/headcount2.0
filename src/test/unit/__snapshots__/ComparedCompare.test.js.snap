// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComparedDistrict unit test suite should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComparedDistricts
    comparedDistricts={
        Array [
            Object {
              "location": "Adams",
              "stats": Object {
                "2004": 0.393,
              },
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <article>
          <h2>
                    hellow
          </h2>
          <p>
                    0.292
          </p>
</article>,
        <p>
          District Data
</p>,
        <article>
          <h2>
                    he
          </h2>
          <p>
                    0.123
          </p>
</article>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "hellow",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              hellow
</h2>,
            <p>
              0.292
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "hellow",
            },
            "ref": null,
            "rendered": "hellow",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 0.292,
            },
            "ref": null,
            "rendered": 0.292,
            "type": "p",
          },
        ],
        "type": "article",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "District Data",
        },
        "ref": null,
        "rendered": "District Data",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "he",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              he
</h2>,
            <p>
              0.123
</p>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "he",
            },
            "ref": null,
            "rendered": "he",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 0.123,
            },
            "ref": null,
            "rendered": 0.123,
            "type": "p",
          },
        ],
        "type": "article",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <article>
            <h2>
                        hellow
            </h2>
            <p>
                        0.292
            </p>
</article>,
          <p>
            District Data
</p>,
          <article>
            <h2>
                        he
            </h2>
            <p>
                        0.123
            </p>
</article>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "hellow",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                hellow
</h2>,
              <p>
                0.292
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "hellow",
              },
              "ref": null,
              "rendered": "hellow",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 0.292,
              },
              "ref": null,
              "rendered": 0.292,
              "type": "p",
            },
          ],
          "type": "article",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "District Data",
          },
          "ref": null,
          "rendered": "District Data",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "he",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2>
                he
</h2>,
              <p>
                0.123
</p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "he",
              },
              "ref": null,
              "rendered": "he",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 0.123,
              },
              "ref": null,
              "rendered": 0.123,
              "type": "p",
            },
          ],
          "type": "article",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
